spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestStructuredStreamingRead3.java

```diff
@@ -206,6 +220,18 @@ public void testReadStreamOnIcebergTableWithMultipleSnapshots_WithNumberOfRows_4
         .isEqualTo(2);
   }
 
+  @TestTemplate
+  public void testReadStreamWithCompositeReadLimit() throws Exception {
+    appendDataAsMultipleSnapshots(TEST_DATA_MULTIPLE_SNAPSHOTS);
+
+    assertThat(
+            microBatchCount(
+                ImmutableMap.of(
+                    SparkReadOptions.STREAMING_MAX_FILES_PER_MICRO_BATCH, "1",
+                    SparkReadOptions.STREAMING_MAX_ROWS_PER_MICRO_BATCH, "2")))
+        .isEqualTo(6);
```

Comment:
This fails without the fix to `SparkMicroBatchStream::getDefaultReadLimit()`, as Spark then calls `SparkMicroBatchStream::latestOffset(Offset, ReadLimit)` with a `CompositeReadLimit` where one of the `ReadLimit`s is a `ReadMaxRows(1)`.